### PR TITLE
[BACK-3055] Add LastDataCutoff and NoData filters to TIDE

### PR DESCRIPTION
**Pull request by Roukoswarf**

### CODE REVIEW

File: api/mappers.go [hunk 1]
```diff
@@ -414,8 +414,7 @@ func NewTideDto(tide *patients.Tide) *Tide {
 			ClinicId:                 &tide.Config.ClinicId,
 			Filters:                  TideFilters(tide.Config.Filters),
 			HighGlucoseThreshold:     tide.Config.HighGlucoseThreshold,
-			LastUploadDateFrom:       tide.Config.LastUploadDateFrom,
```
Comment (toddkazakov) [inline api/mappers.go:417]: Are those parameters removed intentionally?

File: patients/repo.go [hunk 2]
```diff
@@ -1080,6 +1053,51 @@ func cmpToMongoFilter(cmp *string) (string, bool) {
 	return f, ok
 }
 
+func PatientsToTideResult(patientsList []*Patient, period string, exclusions *[]primitive.ObjectID) []TideResultPatient {
```
Comment (toddkazakov) [inline patients/repo.go:1056]: I probably missed this in the previous review, but this type juggling doesn't belong to the persistence layer.

File: xealth_client/gen_client.go [hunk 3]
```diff
@@ -1,7 +1,6 @@
-// WARNING: You are using an OpenAPI 3.1.x specification, which is not yet supported by oapi-codegen (https://github.com/deepmap/oapi-codegen/issues/373) and so some functionality may not be available. Until oapi-codegen supports OpenAPI 3.1, it is recommended to downgrade your spec to 3.0.x
```
Comment (toddkazakov) [inline xealth_client/gen_client.go:1]: This warning was removed by hand. Please fix the makefile so it comments the warning on regenerate.

File: patients/repo.go [hunk 4]
```diff
@@ -1091,16 +1109,8 @@ func (r *repository) TideReport(ctx context.Context, clinicId string, params Tid
...
-	}
-
-	if params.CgmLastUploadDateTo == nil || params.CgmLastUploadDateTo.IsZero() {
-		return nil, errors.New("no lastUploadDateTo provided")
-	}
-
-	if params.CgmLastUploadDateFrom.After(*params.CgmLastUploadDateTo) || params.CgmLastUploadDateFrom.Equal(*params.CgmLastUploadDateTo) {
-		return nil, errors.New("provided lastUploadDateFrom is after or equal to lastUploadDateTo")
+	if params.LastDataCutoff == nil || params.LastDataCutoff.IsZero() {
+		params.LastDataCutoff = pointer.FromAny(time.Now().UTC().AddDate(0, 0, -5))
```
Comment (toddkazakov) [inline patients/repo.go:1113]: Why `-5`?

**Review by toddkazakov — CHANGES_REQUESTED** [review]

File: xealth_client/gen_client.go (hunk 3 above)
Comment (Roukoswarf) [reply xealth_client/gen_client.go:1]: Done

File: patients/repo.go (hunk 2 above)
Comment (Roukoswarf) [reply patients/repo.go:1056]: It was put here due to circular dependency issues, but I can refactor to move more into the api layer I think.

File: patients/repo.go (hunk 4 above)
Comment (Roukoswarf) [reply patients/repo.go:1113]: Was not able to track the reason, so I asked product on intended behavior, and the goal is to make this field required with no default. It now returns an error when unset.

File: patients/repo.go (hunk 2 above)
Comment (toddkazakov) [reply patients/repo.go:1055]: As discussed offline, this code ill be remove in future iterations of the Tide dashboard

File: patients/repo.go [hunk 5]
```diff
@@ -1110,7 +1109,7 @@ func (r *repository) TideReport(ctx context.Context, clinicId string, params Tid
 	tags := store.ObjectIDSFromStringArray(*params.Tags)
 
 	if params.LastDataCutoff == nil || params.LastDataCutoff.IsZero() {
-		params.LastDataCutoff = pointer.FromAny(time.Now().UTC().AddDate(0, 0, -5))
+		return nil, errors.New("no lastDataCutoff provided")
```
Comment (toddkazakov) [inline patients/repo.go:1112]: This will result in a 500. Please wrap `errors.BadRequest`

Comment (Roukoswarf) [reply patients/repo.go]: wrapped.

**Review by clintonium-119 — APPROVED** [review]
LGTM :taco:

**Pull request merged**